mne/inverse_sparse/mxne_inverse.py

```diff
@@ -471,22 +481,21 @@ def tf_mixed_norm(evoked, forward, noise_cov, alpha_space, alpha_time,
     M = np.dot(whitener, M)
 
     # Scaling to make setting of alpha easy
-    n_dip_per_pos = 1 if is_fixed_orient(forward) else 3
     alpha_max = norm_l2inf(np.dot(gain.T, M), n_dip_per_pos, copy=False)
     alpha_max *= 0.01
     gain /= alpha_max
     source_weighting /= alpha_max
 
-    X, active_set, E = tf_mixed_norm_solver(M, gain,
-                                            alpha_space, alpha_time,
-                                            wsize=wsize, tstep=tstep,
-                                            maxit=maxit, tol=tol,
-                                            verbose=verbose,
-                                            n_orient=n_dip_per_pos,
-                                            debias=debias)
+    X, active_set, E = tf_mixed_norm_solver(
+        M, gain, alpha_space, alpha_time, wsize=wsize, tstep=tstep,
+        maxit=maxit, tol=tol, verbose=verbose, n_orient=n_dip_per_pos,
+        log_objective=log_objective, debias=debias)
```

Comment:
That's also someething we should keep in mind for the API change pull request. If we compute the cost function, then why shouldn't we return it. While implementing these funtions I had always versions giving back time and cost function per iteration to check convergence and speed. The problem that I see is that in this case we have two parameters changing the output of the main function, i.e. return_residual and log_objective. So we have to be more careful while calling the function. Maybe theres another more secure idea, e.g. always returing E and residual which are None if return_residual=None and log_objective=None.